debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java

```diff
@@ -74,10 +76,13 @@
     private static final String GET_ALL_CHANGES_FOR_TABLE = "SELECT *# FROM [#db].cdc.[fn_cdc_get_all_changes_#](?, ?, N'all update old') order by [__$start_lsn] ASC, [__$seqval] ASC, [__$operation] ASC";
     protected static final String LSN_TIMESTAMP_SELECT_STATEMENT = "[#db].sys.fn_cdc_map_lsn_to_time([__$start_lsn])";
     protected static final String AT_TIME_ZONE_UTC = "AT TIME ZONE 'UTC'";
-    private static final String GET_LIST_OF_CDC_ENABLED_TABLES = "EXEC [#db].sys.sp_cdc_help_change_data_capture";
+    private static final String GET_LIST_OF_CDC_ENABLED_TABLES = "SELECT s.name AS source_schema, o.name AS source_table, ct.capture_instance, ct.object_id, ct.start_lsn, ct.end_lsn "
```

Comment:
Since we're doing this, it would be nice to have some performance comparison of the previous and the new implementations. Either by profiling or execution plan analysis (`SET SHOWPLAN_ALL ON`).

Reply:
Here is the exec plan of the new query:
```
StmtText                                                                                                                                                                                                                                                       |StmtId|NodeId|Parent|PhysicalOp          |LogicalOp           |Argument                                                                                                                                                                                                                                                       |DefinedValues                                                                                       |EstimateRows|EstimateIO|EstimateCPU|AvgRowSize|TotalSubtreeCost|OutputList                                                                                          |Warnings|Type    |Parallel|EstimateExecutions|
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------+------+------+--------------------+--------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------+------------+----------+-----------+----------+----------------+----------------------------------------------------------------------------------------------------+--------+--------+--------+------------------+
SELECT ¶ s.name AS source_schema,¶ o.name AS source_table,¶ ct.capture_instance ,¶ ct.object_id ,¶ ct.start_lsn ,¶ ct.end_lsn¶FROM cdc.change_tables ct ¶LEFT JOIN sys.objects o ON ct.source_object_id = o.object_id ¶LEFT JOIN sys.schemas s ON s.schema_id =|     1|     1|     0|                    |                    |1                                                                                                                                                                                                                                                              |                                                                                                    |   2.2394173|          |           |          |     0.010187132|                                                                                                    |        |SELECT  |       0|                  |
  |--Nested Loops(Left Outer Join, OUTER REFERENCES:([o].[nsid]))                                                                                                                                                                                              |     1|     2|     1|Nested Loops        |Left Outer Join     |OUTER REFERENCES:([o].[nsid])                                                                                                                                                                                                                                  |                                                                                                    |   2.2394173|       0.0| 0.00000838|       423|     0.010187132|[ct].[object_id], [ct].[capture_instance], [ct].[start_lsn], [ct].[end_lsn], [o].[name], [s].[name] |        |PLAN_ROW|       0|               1.0|
       |--Nested Loops(Left Outer Join, OUTER REFERENCES:([ct].[source_object_id]))                                                                                                                                                                            |     1|     3|     2|Nested Loops        |Left Outer Join     |OUTER REFERENCES:([ct].[source_object_id])                                                                                                                                                                                                                     |                                                                                                    |    2.004021|       0.0| 0.00000836|       297|      0.00673692|[ct].[object_id], [ct].[capture_instance], [ct].[start_lsn], [ct].[end_lsn], [o].[name], [o].[nsid] |        |PLAN_ROW|       0|               1.0|
       |    |--Clustered Index Scan(OBJECT:([testDB1].[cdc].[change_tables].[change_tables_clustered_idx] AS [ct]))                                                                                                                                            |     1|     4|     3|Clustered Index Scan|Clustered Index Scan|OBJECT:([testDB1].[cdc].[change_tables].[change_tables_clustered_idx] AS [ct])                                                                                                                                                                                 |[ct].[object_id], [ct].[source_object_id], [ct].[capture_instance], [ct].[start_lsn], [ct].[end_lsn]|         2.0|  0.003125|  0.0001592|       167|       0.0032842|[ct].[object_id], [ct].[source_object_id], [ct].[capture_instance], [ct].[start_lsn], [ct].[end_lsn]|        |PLAN_ROW|       0|               1.0|
       |    |--Filter(WHERE:(has_access('CO',[testDB1].[sys].[sysschobjs].[id] as [o].[id])=(1)))                                                                                                                                                              |     1|     5|     3|Filter              |Filter              |WHERE:(has_access('CO',[testDB1].[sys].[sysschobjs].[id] as [o].[id])=(1))                                                                                                                                                                                     |                                                                                                    |         1.0|       0.0| 0.00000158|       143|      0.00344436|[o].[name], [o].[nsid]                                                                              |        |PLAN_ROW|       0|               2.0|
       |         |--Clustered Index Seek(OBJECT:([testDB1].[sys].[sysschobjs].[clst] AS [o]), SEEK:([o].[id]=[testDB1].[cdc].[change_tables].[source_object_id] as [ct].[source_object_id]),  WHERE:([testDB1].[sys].[sysschobjs].[nsclass] as [o].[nsclass]=(0|     1|     6|     5|Clustered Index Seek|Clustered Index Seek|OBJECT:([testDB1].[sys].[sysschobjs].[clst] AS [o]), SEEK:([o].[id]=[testDB1].[cdc].[change_tables].[source_object_id] as [ct].[source_object_id]),  WHERE:([testDB1].[sys].[sysschobjs].[nsclass] as [o].[nsclass]=(0) AND [testDB1].[sys].[sysschobjs].[pclas|[o].[id], [o].[name], [o].[nsid]                                                                    |         1.0|  0.003125|  0.0001581|       149|       0.0034412|[o].[id], [o].[name], [o].[nsid]                                                                    |        |PLAN_ROW|       0|               2.0|
       |--Clustered Index Seek(OBJECT:([testDB1].[sys].[sysclsobjs].[clst] AS [s]), SEEK:([s].[class]=(50) AND [s].[id]=[testDB1].[sys].[sysschobjs].[nsid] as [o].[nsid]) ORDERED FORWARD)                                                                    |     1|     7|     2|Clustered Index Seek|Clustered Index Seek|OBJECT:([testDB1].[sys].[sysclsobjs].[clst] AS [s]), SEEK:([s].[class]=(50) AND [s].[id]=[testDB1].[sys].[sysschobjs].[nsid] as [o].[nsid]) ORDERED FORWARD                                                                                                    |[s].[name]                                                                                          |         1.0|  0.003125|  0.0001581|       139|    0.0034418358|[s].[name]                                                                                          |        |PLAN_ROW|       0|          2.004021|
```
The exec plan of the stored procedure:
[exec sys.sp_cdc_help_change_data_capt.txt](https://github.com/sugarcrm/debezium/files/6510460/exec.sys.sp_cdc_help_change_data_capt.txt)

I don't know how to compare these, but obvously the first one is way simplier.

Reply:
> I don't know how to compare these, but obvously the first one is way simplier.

This must be sufficient to start the discussion when contributing this back upstream. Thanks!